.eslintignore file has no effect

I'm on version `6.0.0`. I have a directory structure like this:

```
js/
- node_modules/
- lib/
  - gen/
- .eslintignore
- .eslintrc.json
- .gitignore
- package.json
```

My `.eslintignore` looks like this:
```
node_modules/
lib/gen/
```

And my `.gitignore` file looks like this:
```
lib/gen/
```

(I don't .gitignore my node_modules)

In Atom with `linter-eslint@6.0.0`, opening any `.js` files inside my `node_modules` directory will result in half of my screen being bombarded with lint errors. Opening any `.js` files inside the `lib/gen` directory will not. Furthermore, if I remove `lib/gen` from my `.eslintignore` file, the files inside that directory still do not get linted by the text editor. Only when I remove the path from my `.gitignore` does the linter actually activate for files in that directory.

With some investigation, I added a `console.log()` statements above line 149 of `linter-eslint/lib/main.js` and discovered that opening any files that are under `.gitignore` don't ever even trigger the worker job request.

Well, just came up with a test case for this. As you can see in the picture, foo.js shouldn't be linted as it's in the `.eslintignore` file. The files for the testcase can be found here: https://gist.github.com/peregrinogris/929293d7f37d9e42a75a
![linter-eslint error](http://i.imgur.com/yCdvpPW.png)
When I try to lint that file from the console, I get:
```
$ ./node_modules/.bin/eslint foo.js

/Users/hcolmeiro/test/eslint-error/foo.js
  0:0  warning  File ignored because of a matching ignore pattern. Use --no-ignore to override

✖ 1 problem (0 errors, 1 warning)
```

--- a/lib/worker-helpers.js
+++ b/lib/worker-helpers.js
@@ -135,6 +135,11 @@ function getArgv(type, config, filePath, fileDir, givenConfigPath) {
   }
   argv.push('--format', _path2.default.join(__dirname, 'reporter.js'));
 
+  var ignoreFile = config.disableEslintIgnore ? null : (0, _atomLinter.findCached)(fileDir, '.eslintignore');
+  if (ignoreFile) {
+    argv.push('--ignore-path', ignoreFile);
+  }
+
   if (config.eslintRulesDir) {
     var rulesDir = (0, _resolveEnv2.default)(config.eslintRulesDir);
     if (!_path2.default.isAbsolute(rulesDir)) {

--- a/src/worker-helpers.js
+++ b/src/worker-helpers.js
@@ -119,6 +119,11 @@ export function getArgv(type, config, filePath, fileDir, givenConfigPath) {
   }
   argv.push('--format', Path.join(__dirname, 'reporter.js'))
 
+  const ignoreFile = config.disableEslintIgnore ? null : findCached(fileDir, '.eslintignore')
+  if (ignoreFile) {
+    argv.push('--ignore-path', ignoreFile)
+  }
+
   if (config.eslintRulesDir) {
     let rulesDir = resolveEnv(config.eslintRulesDir)
     if (!Path.isAbsolute(rulesDir)) {

